src/plugin/luapi_application.h

```diff
@@ -867,6 +876,132 @@ static int applib_getDisplayDpi(lua_State* L) {
 }
 
 
+/**
+ * Exports the current document as a pdf or as a svg or png image
+
+ * Example 1:
+ * app.export({["outputFile"] = "Test.pdf", ["range"] = "2-5; 7", ["background"] = "none", ["progressiveMode"] = true})
+ *
+ * Example 2:
+ * app.export({["outputFile"] = "Test.svg", ["range"] = "3-", ["background"] = "unruled"})
+ *
+ * Example 3:
+ * app.export({["outputFile"] = "Test.png", ["range"] = "1-2", ["background"] = "all", ["pngWidth"] = 800})
+ **/
+static int applib_export(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    Document* doc = control->getDocument();
+
+    // discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+
+    lua_getfield(L, 1, "outputFile");
+    lua_getfield(L, 1, "range");
+    lua_getfield(L, 1, "background");
+    lua_getfield(L, 1, "progressiveMode");
+    lua_getfield(L, 1, "pngDpi");
+    lua_getfield(L, 1, "pngWidth");
+    lua_getfield(L, 1, "dpiHeight");
+
+    const char* outputFile = luaL_optstring(L, -7, nullptr);
+    const char* range = luaL_optstring(L, -6, nullptr);
+    const char* background = luaL_optstring(L, -5, "all");
+    bool progressiveMode = lua_toboolean(L, -4);  // true unless nil or false
+    int pngDpi = luaL_optinteger(L, -3, -1);
+    int pngWidth = luaL_optinteger(L, -2, -1);
+    int pngHeight = luaL_optinteger(L, -1, -1);
+
+    ExportBackgroundType bgType = EXPORT_BACKGROUND_ALL;
+    if (strcmp(background, "unruled") == 0) {
+        bgType = EXPORT_BACKGROUND_UNRULED;
+    } else if (strcmp(background, "none") == 0) {
+        bgType = EXPORT_BACKGROUND_NONE;
+    }
+
+    if (outputFile == nullptr) {
+        luaL_error(L, "Missing output file!");
+    }
+
+    fs::path file = fs::path(outputFile);
+    auto extension = file.extension();
+
+    if (extension == ".pdf") {
+        // exportPdf("test.xopp", outputFile, range, bgType, progressiveMode);
+        GFile* file = g_file_new_for_commandline_arg(outputFile);
+
+        XojPdfExport* pdfe = XojPdfExportFactory::createExport(doc, nullptr);
```

Comment:
We're deleting `XojPdfExport` soon after we're creating it. Let's consider using a `std::unique_ptr` here!

Reply:
Done